### PR TITLE
Fix gitserver bc search

### PR DESCRIPTION
**Pull request by csrwng**
Fixes bug in gitserver search for matching build config (pointer to loop var).
Adds unit test to catch the bug.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees ptal

File: pkg/gitserver/getbuildconfigs.go [hunk 1]
```diff
@@ -14,30 +14,27 @@ import (
...
 	ns := os.Getenv("POD_NAMESPACE")
-	buildConfigList, err := client.BuildConfigs(ns).List(kapi.ListOptions{})
+	buildConfigList, err := c.BuildConfigs(ns).List(kapi.ListOptions{})
 	if err != nil {
 		return err
 	}
 
 	matchingBuildConfigs := []*buildapi.BuildConfig{}
 
-	for _, bc := range buildConfigList.Items {
```
Comment (bparees) [inline pkg/gitserver/getbuildconfigs.go]: go's horrible iteration behavior strikes again, eh?

Comment (csrwng) [inline pkg/gitserver/getbuildconfigs.go]: yes :)

**Comment by bparees** [comment]
lgtm [merge]

**Comment by csrwng** [comment]
https://github.com/openshift/origin/issues/8221
[merge]

**Comment by csrwng** [comment]
[test]

**Comment by bparees** [comment]
@stevekuznetsov why isn't the bot respecting @csrwng's flake comment? https://github.com/openshift/origin/pull/10576#issuecomment-241715595

[merge]

**Comment by bparees** [comment]
@stevekuznetsov nm, i guess the issue @csrwng linked didn't have the flake label.

**Pull request merged**